### ISSUE TITLE
local documentation format is wrong in konqueror

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4812]
|<img alt="douglas's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| douglas reported|
|-|-|
|Trac ID|trac#224|
|Type|defect|
|Created|2003-08-22 18:22:19Z|

<details><summary>Searchable metadata</summary>

```
trac-id__224 224
type__defect defect
reporter__douglas douglas
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__documentation documentation
time__1061576539000000 1061576539000000
changetime__1468641600850323 1468641600850323
version__ 
owner__ 
cc__radix cc__Nafai cc__spiv cc__alexander cc__moshez cc__douglas
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="douglas's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|douglas commented|
|-|-|

```
#!html
<pre>
In Konqueror (3.1.3), the local version of
http://www.twistedmatrix.com/documents/howto/pageindepth
(ie file:/usr/share/doc/twisted-doc/howto/pageindepth.xhtml)
on debian sid (with 1.0.7a5), the examples are not displayed
as preformatted text, rather as one long unpythonic line. 

there is no obvious flaw in the source. 

(I would not have noticed, but my mozilla is playing up and
the online docs are slow today).

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Nafai's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Nafai commented|
|-|-|

```
#!html
<pre>
I would add that the files in twisted-doc cannot be viewed
with elinks or lynx.  I try viewing them with one of the
above  and it thinks I want to download them.  I believe a
friend had the same problem with Opera

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
Nafai: The problem with elinks/lynx is that they don't
recognise local .xhtml files as HTML.  There was a thread
about this on the mailing list several months ago; initially
we guessed that the lack of !DOCTYPE declarations was the
problem, but no, links simply required that the docs have
.html as the extension, which we won't do for various
reasons.  If you view the exact same docs via a webserver
you don't have this problem, because of the HTTP header
"Content-type: text/html" that gets set, so a workaround is
to install apache or twisted-web, and read the docs via
http://localhost/doc (debian webservers provide that by
default).

douglas: I don't have Konquerer installed (although I could
grab it if I had to), but the HTML/CSS looks fine at a
glance.  Perhaps the new CSS 2.1 draft has something that
would help here?  Anyway, without a suggestion of how to fix
the HTML to make Konq happy, this probably won't get fixed
(I guess that's why you filed it as a "wish" :).

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
I don't remember the reasons that we won't distribute our
docs as .html. what was the thread that this was stated in?

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
radix: The thread had the subject: 
    Twisted 1.0.4 - `html' .vs. `xhtml'

Although skimming it again, I don't see any particular
reasons why we don't distribute them as .html, except that
it shouldn't matter and links sucks ;)

Regardless, this is unrelated to the original bug report
from douglas -- perhaps this discussion should be moved to a
fresh bug?

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Not working in standard browsers is a bug, not a wish :)

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
I'm assigning it to spiv -- this is mostly an issue of Lore generation.
Andrew, if you feel it doesn't belong to you, feel free to unassign.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
There is a reason why not to distribute it as .html, but it might not be
a "good one". The issue here is that the source files are *ALSO* named .html,
which is kind of a pain. One way to solve this would be to support --outdir
in Lore, and put everything in a parallel directory structure...except that
image location considerations need to be kept in mind. Another alternative
would be to have a "great renaming" of documents to .xhtml, and to produce
.html from them. Pursuing the latter option seems to be the most work, but
the least brain-power. If anyone wishes to do it, the places to touch are
doc/{howto,specifications,vision}, doc/index.html, and admin/process-documents.
Then debian/rules will need to be touched -- but there it's a fairly mindless
s/xhtml/html [I will do this]

Any comments?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Fixed.
Moved all sources to .xhtml, moved targets to .html.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="alexander's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|alexander commented|
|-|-|

```
#!html
<pre>
The original issue was not resolved—all of the documentation need document
types, so I am adding them as necessary.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Um, no.
The original issue was a mimetype, not a doctype, problem.
I solved the mimetype problem. Please do not reopen unless
the docs *don't work in konqueror*.
Thanks
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_13"></a>|GitHub commented|
|-|-|

In changeset c5a4c635de259cd1b92a555c930aa426164f9cce
```
#!CommitTicketReference repository="" revision="c5a4c635de259cd1b92a555c930aa426164f9cce"
Merge pull request #224 from twisted/8491-rodrigc-console

Author: rodrigc
Reviewer: alex, adiroiban
Fixes: #8491

Convert scripts in bin to console_scripts
```
